client/src/connection_cache.rs

```diff
@@ -90,7 +93,19 @@ impl ConnectionCache {
         }
         if let Some(client_socket) = client_socket {
             config.update_client_endpoint(client_socket);
+        } else if cfg!(debug_assertions) {
```

Comment:
I think a better place do this is in https://github.com/anza-xyz/agave/blob/master/quic-client/src/nonblocking/quic_client.rs#L83 so that we can still test the logic of delayed endpoint creation same as production code.

Reply:
That place already binds outside of range used for CI test range allocators.

Reply:
see https://github.com/anza-xyz/agave/pull/7801#issuecomment-3238142858